ConwaysGameOfLife/GameController.cs

```diff
@@ -42,19 +42,57 @@ public GameController(IInput input, IOutput output)
         public void Run()
         {
             _output.WriteLine(Messages.Welcome);
-            CreateInitialUniverse();
+            _output.WriteLine(Messages.RequestDimensions);
+            var input = _input.ReadLine();
+            
+            if (UserEndsGame(input)) return;
+
+            var isValidUniverse = Validator.IsValidUniverse(input);
+            while (!isValidUniverse)
+            {
+                _output.WriteLine(Messages.InvalidInput);
+                _output.WriteLine(Messages.RequestDimensions);
+                input = _input.ReadLine();
+                isValidUniverse = Validator.IsValidUniverse(input);
+            }

```

Comment:
what happens when the user inputs `"q"` here? lets write a test for that

Reply:
At which point does the user have the option to quit the application - is it immediately after the welcome message is displayed and before requesting the dimensions, or only after the set up is finished? (up to you - but this may affect the structure of your loop)